### ISSUE TITLE
Provide instructions for 'docker-in-docker' for docker based planemo machine

### ISSUE THREAD

**Issue by kellrott** [galaxyproject/planemo-machine#26]
Instructions should be added to the deployment of the docker image of the planemo box to instruct the user how to mount the parent OS docker server so the box can deploy sub-containers.

I've used:
--privileged -v /var/run/docker.sock:/var/run/docker.sock

Which mounts in socket to the parent OS's docker server (still need to verify that privileged is necessary)

**Comment by jmchilton** [comment]
Yes it should - this command works on Mac OS X?

**Comment by kellrott** [comment]
Weirdly it does. boot2docker only remounts the users home directory into the running VM. So the '/var/run/docker.sock' path that gets posted in isn't from the Mac OS, but rather the VM OS.

**Comment by bgruening** [comment]
@kellrott OS-X can never have a /var/run/docker.sock, isn't it?

**Comment by kellrott** [comment]
Yeah, /var/run/docker.sock doesn't exist in MacOS.  Boot2Docker's 'shellinit' command sets the DOCKER_HOST environmental variable, which gets used rather then '/var/run/docker.sock'. But if you issue a docker run with '-v /var/run/docker.sock:/var/run/docker.sock' in MacOS, the path gets forwarded into the VM, where it hits a socket that does exist.
